bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusSlave.java

```diff
@@ -198,16 +198,15 @@ private void setCoil(Command command, ModbusBindingConfig config)
      * @throws ModbusException ModbusIOException on IO errors, ModbusSlaveException with protocol level exceptions
      * @throws ModbusUnexpectedTransactionIdException when response transaction id does not match the request
      */
-    protected void setRegister(Command command, ModbusBindingConfig config)
+    protected void setRegister(Command command, int readIndex, int writeRegister)
             throws ModbusConnectionException, ModbusException, ModbusUnexpectedTransactionIdException {
-        int readIndex = config.readIndex;
-        int writeRegister = getStart() + config.writeIndex;
+        writeRegister += getStart();
 
         Register newValue;
         if (command instanceof IncreaseDecreaseType) {
             newValue = readCachedRegisterValue(readIndex);
             if (newValue == null) {
-                logger.warn("Not polled value for item {}. Cannot process command {}", config.getItemName(), command);
+                logger.warn("Not polled value. Cannot process command {}", command);

```

Comment:
My 1st approach was w/o the config param in this call to keep the API small. 
Since it is back I can revert this.